src/librustc/lint/builtin.rs

```diff
@@ -320,17 +320,18 @@ impl LintPass for HardwiredLints {
 #[derive(PartialEq, RustcEncodable, RustcDecodable, Debug)]
 pub enum BuiltinLintDiagnostics {
     Normal,
-    BareTraitObject(Span)
+    BareTraitObject(Span, /* is_global */ bool)
 }
 
 impl BuiltinLintDiagnostics {
     pub fn run(self, sess: &Session, db: &mut DiagnosticBuilder) {
         match self {
             BuiltinLintDiagnostics::Normal => (),
-            BuiltinLintDiagnostics::BareTraitObject(span) => {
-                let sugg = match sess.codemap().span_to_snippet(span) {
-                    Ok(s) => format!("dyn {}", s),
-                    Err(_) => format!("dyn <type>")
+            BuiltinLintDiagnostics::BareTraitObject(span, is_global) => {
+                let sugg = match (sess.codemap().span_to_snippet(span), is_global) {
+                    (Ok(s), true) => format!("dyn ({})", s),
```

Comment:
Nit: this would (imo) be slightly easier to read as

```rust
let sugg = match sess.codemap().span_to_snippet(span) {
    Ok(s) if is_global => ...
    Ok(s) => ...
    _
}
```

Reply:
We do that, perhaps you're reviewing an earlier commit?